### PR TITLE
Version validation should use the original NuGet.Core regex to ensure compat #3226

### PR DESCRIPTION
**Pull request by maartenba**
Fixes https://github.com/NuGet/NuGetGallery/issues/3226

@skofman1 @scottbommarito @ryuyu @xavierdecoster 

@emgarten can you have a look as well if this is the intended course?

### CODE REVIEW

**Comment by xavierdecoster** [comment]
LGTM

File: src/NuGetGallery.Core/Packaging/ManifestValidator.cs [hunk 1]
```diff
@@ -143,11 +141,47 @@ private static IEnumerable<ValidationResult> ValidateCore(PackageMetadata packag
...
+                            {
+                                yield return validationResult;
+                            }
+                        }
                     }
                 }
             }
         }
 
+        private static IEnumerable<ValidationResult> ValidateVersion(NuGetVersion version)
```
Comment (skofman1) [inline src/NuGetGallery.Core/Packaging/ManifestValidator.cs]: > IEnumerable<ValidationResult> [](start = 22, length = 30)

A bit confused: from what I understand we always return a single ValidationResult, so why we need an IEnumerable here? What did I get wrong?

Comment (maartenba) [inline src/NuGetGallery.Core/Packaging/ManifestValidator.cs]: Refactoring leftover... Good catch!

**Comment by skofman1** [comment]
:shipit:

File: src/NuGetGallery.Core/NuGetVersionExtensions.cs [hunk 2]
```diff
@@ -32,5 +36,12 @@ public static bool IsSemVer200(this NuGetVersion self)
         {
             return self.ReleaseLabels.Count() > 1 || self.HasMetadata;
         }
+
+        public static bool IsValidVersion(this NuGetVersion self)
```
Comment (emgarten) [inline src/NuGetGallery.Core/NuGetVersionExtensions.cs]: IsValidVersionForLegacyClients maybe? :laughing:

**Comment by emgarten** [comment]
Looks great, thanks for fixing this! :shipit:

**Pull request merged**